### PR TITLE
fix: new component InputNumber created 🧩

### PR DESCRIPTION
**Pull request by aktasmehmet**
## Description

- This PR moves the current number type related code out of Motif/InputText and creates a brand new InputNumber component with more features. 

- This component basically mimics a text typed input component to look like a number typed one.

- type prop is removed from Motif/InputText

- Test are added coverint all possible scenarios and edge cases. Form related tests are also added or updated.

- GlobalIconWrapper is also updated because of the icon usage of InputNumber

**There are also some side code changes here:**
- Created a common InputText type to share between InputText, InputPassword and InputNumber. This leads adding some new props to InputPassword such as clearable, etc.
- Based on the change above, type, story and test changes applied
- icon prop of InputPassword has replaced with iconLeft

## Type of Change

- [x] 🆕 New Component
- [x] ✨ Feature / Enhancement
- [ ] 🐛 Bug Fix
- [ ] 💥 Breaking Change
- [ ] 📦 Build / Dependency / Docs Update
- [x] 🧹 Code Refactoring

## Screenshots / Preview

Screenshot 2026-06-29 at 15 57 28

## Checklist

- [x] Self-reviewed, no leftover logs or debug code
- [x] Uses design tokens — no hardcoded style values
- [ ] Accessible (keyboard nav, ARIA, contrast)
- [x] Tests added/updated and passing
- [x] Storybook story added/updated

## How to Test

Just use the component...

### CODE REVIEW

File: src/lib/components/InputNumber/types.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+export type InputNumberProps = {
+  value?: number;
+  min?: number;
+  max?: number;
+} & Omit<InputCommonProps, "value"> &
+  InputTextCommonProps &
+  InputNumberDefaultableProps;
+
+export type InputNumberDefaultableProps = {
```
[the remark this answers is not in the document]
Comment (aktasmehmet) [reply src/lib/components/InputNumber/types.ts:14]: InputNumberDefaultableProps is designed to be used in the context. So it has to be there. I will omit some those props from InputTextCommonProps there to prevent duplication.

File: src/lib/components/Motif/InputText/components/NumberSpinner.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  max?: number;
+  onChange: (value: string, updateInputRefValue?: boolean) => void;
+  disabled?: boolean;
+};
+
+const NumberSpinner = (props: Props) => {
+  const { value, min, max, onChange, disabled } = props;
+  const adjustNumberValue = useCallback(
+    (v: number) => {
+      const raw = (Number(value) || 0) + v;
```
[the remark this answers is not in the document]
Comment (atakanbayrak) [reply src/lib/components/Motif/InputText/components/NumberSpinner.tsx]: String to number consistency seems to be solved with previous 2 comments comes from Claude.

File: src/lib/components/InputNumber/InputNumber.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  const classNames = sanitizeModuleRootClasses(styles, props.className);
+
+  return (
+    <InputText
+      {...props}
+      {...propsFromForm}
+      value={props.value !== undefined ? String(props.value) : undefined}
+      onValueUpdated={onFormFieldValueUpdate}
+      valueTransformer={filterValue}
+      onBlur={handleBlur}
```
[the remark this answers is not in the document]
Comment (aktasmehmet) [reply src/lib/components/InputNumber/InputNumber.tsx:89]: Are you sure about that? Onblur is not an exposed prop of InputNumber.

File: src/lib/components/Motif/InputText/InputText.tsx [hunk 4]
```diff
@@ -126,15 +145,14 @@ const InputText = (props: PropsWithRef<InternalInputProps, HTMLDivElement>) => {
...
-          </button>
-        </div>
+      {numberSpinner && (
+        <NumberSpinner
+          value={itemValue}
+          min={numberSpinner.min}
+          max={numberSpinner.max}
+          disabled={disabled || readOnly}
+          onChange={changeProcess}
+        />
```
Comment (atakanbayrak) [inline src/lib/components/Motif/InputText/InputText.tsx:156]: InternalInputText components supports numberSpinner but normal InputText component does not have that type. Should be added and spread inside props.

File: src/lib/motif/types/componentDefaultableProps.ts [hunk 5]
```diff
@@ -44,7 +45,6 @@ import { RadioGroupDefaultableProps } from "../../components/RadioGroup/types";
 import { SelectDefaultableProps } from "../../components/Select/types";
 import { SliderDefaultableProps } from "../../components/Slider/types";
 import { SliderRangeDefaultableProps } from "../../components/SliderRange/types";
-// import { SkeletonDefaultableProps } from "../../components/Skeleton/types";
```
Comment (atakanbayrak) [inline src/lib/motif/types/componentDefaultableProps.ts:47]: If we delete this import, we might consider deleting SkeletonDefaultableProps either.

File: src/lib/components/Motif/InputText/InputText.tsx (hunk 4 above)
Comment (aktasmehmet) [reply src/lib/components/Motif/InputText/InputText.tsx:156]: Nope, it is removed intentionally. InputText is now only for text. Number spinner is not part of it.

Comment (atakanbayrak) [reply src/lib/components/Motif/InputText/InputText.tsx:156]: mb 😢

**Review by atakanbayrak — DISMISSED** [review]
I have nothing to add, when Skeleton issue is solved or ignored good to go.

File: src/lib/components/InputNumber/helper.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    allowDecimals && decimalScale !== undefined && dotIndex !== -1 ? decimalFixed.slice(0, dotIndex + 1 + decimalScale) : decimalFixed;
+
+  // 4. Fix misplaced minus signs
+  const signFixed = allowNegative ? (scaleFixed.startsWith("-") ? "-" : "") + scaleFixed.replace(/-/g, "") : scaleFixed;
+
+  // 5. Enforce maxLength boundary
+  const lengthFixed = maxLength !== undefined && signFixed.length > maxLength ? signFixed.slice(0, maxLength) : signFixed;
+
+  // 6. Reject if all characters were invalid (e.g. pasting "xxx" into a non-empty field)
+  if (lengthFixed === "" && value !== "") return undefined;
```
[the remark this answers is not in the document]
Comment (aktasmehmet) [reply src/lib/components/InputNumber/helper.ts:34]: The paste reversal behavior is correct design: pasting "xxx" into a non-empty field restores the previous value rather than silently becoming empty. Debatable UI-wise, but it's consistent with the rejection 
  contract of valueTransformer. Worth keeping as-is.

[the remark this answers is not in the document]
Comment (aktasmehmet) [reply src/lib/components/InputNumber/helper.ts:34]: !isNaN guard intentionally lets incomplete values like "-" and "-." pass through while the user is still typing.

**Review by ZehranurC — APPROVED** [review]
